src/metabase/lib/schema.cljc

```diff
@@ -144,8 +144,8 @@
     {:error/message ":source-query is not allowed in pMBQL queries."}
     #(not (contains? % :source-query))]
    [:fn
-    {:error/message "A query cannot have both a :source-table and a :source-card."}
-    (complement (every-pred :source-table :source-card))]
+    {:error/message "A query can only have one of :source-table, :source-card, and :sources"}
```

Comment:
I'd make this wording even stronger, to match the condition below: `A query must have exactly one of :source-table, :source-card, or :sources`